clean_project/search.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+"""
+Search a word in a file and return all the occurrence specified the line numbers
+"""
+import argparse
+import json
+
+BOLD = '\033[1m'
+END = '\033[0m'
+
+
+def search(search_word, json_path):
+    """
+    Print all the occurrence of a given word
+    :param search_word: Word given to search
+    :param json_path: Indexing file
+    :return: List of files and numbers of file when search_word occurrence
+    """
+    with open(json_path) as json_file:
+        list_of_dict = json.load(json_file)
+        count_match = 0
+        result_list = []
+        for file_dict in list_of_dict:
+            if search_word in file_dict:
+                count_match += 1
+                print "The word " + BOLD + search_word + END + " is contains in the following file/lines:"
+                for occurrence in file_dict.get(search_word):
+                    file_name = file_dict.get('file_name')[0].split('/')
+                    print "--> " + file_name[len(file_name)-1] + ': ' + str(occurrence)
+                    result_list.append("--> " + file_name[len(file_name)-1] + ': ' + str(occurrence))
```

Comment:
don't see much point in returning the result list if you're printing it. you may say that in this way you can test it properly, but then i ask you why printing in this function then. if you return a list of results, that's fair enough and actually it's the best option you have in my opinion, but you shouldn't be printing here. you should have a presentation function whose only responsibility is taking in an input of occurrences (result_list) and print them properly. this function should be called not by `search()` because it's not its responsibility to think to the presentation

also the result list should return only the occurrences and not formatted strings